app/src/actioncommands.cpp

```diff
@@ -563,6 +568,186 @@ void ActionCommands::duplicateKey()
     mEditor->layers()->notifyAnimationLengthChanged();
 }
 
+
+void ActionCommands::manipulateRange()
+{
+    int loopBegin = 1, loopEnd = 2; // default values
+    if (mEditor->playback()->isRangedPlaybackOn())
+    {   // defined range values if available
+        loopBegin = mEditor->playback()->getRangedStartFrame();
+        loopEnd = mEditor->playback()->getRangedEndFrame();
+    }
+    LayerManager* layerMgr = mEditor->layers();
+    int lIndex = layerMgr->currentLayer()->type();
+    //   BITMAP = 1     VECTOR = 2  //not yet      SOUND = 4     CAMERA = 5
+    if (lIndex == 1 || lIndex == 2) //not yet || lIndex == 4 || lIndex == 5)
```

Comment:
The "dozens of lines" are the heart of the function. I can't see your idea...?

Reply:
What I’m saying is that you could turn this:

```cpp
void foo()
{
    if (bar) {
        // dozens of lines
    } else {
        baz()
    }
}
```

into this:

```cpp
void foo()
{
    if (!bar) {
        baz();
        return;
    }
    // dozens of lines, now no longer wrapped by if statement
}
```

And looking back, I agree that my original comment was worded rather confusingly.

Reply:
Ok. Now it makes sense.

Reply:
Could you clarify how this is even possible? When I put a break in an if, I get “error: 'break' statement not in loop or switch statement”. Even with that aside I have no idea how you would run into a situation where both sections are executed, unless you’re deliberately working with goto or something to do achieve just that (and goto is pretty evil to begin with).

That aside, it’s true that this pattern is aimed at cases where the if serves as a “guard” to execute code only if a certain condition holds true and otherwise run just some clean-up or error-handling or whatever instead. But if you got different things to do for each possible value of a particular set you should probably be using switch anyway.

Reply:
Yeah you're right, I don't know what I was thinking. Breaks don't work on if statements. I guess this is just done by convention and for reducing indentation.

Reply:
After thinking about this some more, I guess you intended to refer to a case where all return statements within the if are skipped. That can indeed prove a problem, but it’s essentially the same problem as a missing break in a switch statement. In that regard, it could also be used for an intentional fall-through, though (but admittedly the cases wouldn’t be entirely mutually exclusive, then). But I should probably have some sleep now because thinking clearly is slowly starting becoming more difficult.

Reply:
I've negated the condition as suggested in the first post.